packages/cli-repl/src/cli-repl.ts

```diff
@@ -163,6 +163,10 @@ class CliRepl {
           cs.password = await this.requirePassword();
           driverUri = cs.href;
         }
+        if (this.isAppNameMissingURI(cs)) {
+          cs.searchParams.set('appName', `mongosh ${version}`);
+          driverUri = cs.href;
+        }
```

Comment:
If this `if` remains here, if wouldn’t be triggered in the cases in which we prompt the user for password (line 159 above, happens with e.g. `mongosh --host localhost --username user`).